DOC: Develop-3 Documentation Update (#524)
pysat/pysat: docs/installation.rst

   pip install pysat

> Note that pysat requires a number of packages for the install.  The upper caps
> for packages below have been removed for the upcoming pysat 3.0.0 release.
> The development version can be found in the ``develop-3``
> branch (see above for caveats).
> 
>      * beautifulsoup4
>      * h5py
>      * lxml
>      * madrigalWeb
>      * matplotlib
>      * netCDF4
>      * numpy (>=1.12)
>      * pandas (>=0.23, <0.25)
>      * PyForecastTools
>      * pysatCDF
>      * requests
>      * scipy
>      * xarray (<0.15)
> 
> The upper caps for packages above will be removed for the upcoming pysat
> 3.0.0 release.

Reviewer: ```suggestion
There are a few packages that pysat depends on, and need to be installed:

     * netCDF4
     * numpy (>=1.12)
     * pandas (>=0.23)
     * scipy
     * xarray

```

Does pysat install these itself if you use `python setup.py install`?

Author: Yes.

**Development Installation**

----

pysat may also be installed directly from the source repository on github::

   git clone https://github.com/pysat/pysat.git
   cd pysat
   python setup.py install

An advantage to installing through github is access to the development branches.
The latest bugfixes can be found in the ``develop`` branch. However, this
branch is not stable (as the name implies). We recommend using this branch in a
virtual environment and using::

   git clone https://github.com/pysat/pysat.git
   cd pysat
   git checkout develop
   python setup.py develop

The use of `develop` rather than `install` installs the code 'in-place', so
any changes to the software do not have to be reinstalled to take effect.

> The development version for v3.0 can be found in the ``develop-3``
> branch (see above for caveats).

Reviewer: ```suggestion
```

**Windows**

----

To get pysat installed in Windows you need a POSIX compatible C/ Fortran
compiling environment. This is required to compile the
`pysatCDF <https://github.com/pysat/pysatCDF/>`_ package.

> Python environment: Python 2.7.x

Reviewer: This should be updated.

Author: I passed a note to @mddepew who is the original author of that text to see if there are any updates needed for Python 3 on windows, other than that version text there.

Author: Updated.

#. Next step is to add the mingw64 bin folder to your windows system path. Right-click on computer, hit properties. Then click advanced system settings, then environment variables. Find the system variable (as opposed to user variables) named PATH. This is a semicolon delimited list of the OS search paths for binaries. Add another semicolon and the path ``C:\msys64\mingw64\bin``
#. Now you should have access to Python from within your MSYS2 environment. And your windows path should have access to the mingw binaries. To verify this, launch the mingw64 MSYS2 environment.::

    C:\msys64\mingw64.exe

   Run the command::

    which python

   and confirm that it points to the correct python version you want to be using.
#. Microsoft Visual C++ 9.0 is required to compile C sources. Download and
   install the right version of Microsoft Visual C++ for Python 2.7
   from `<http://aka.ms/vcpython27>`_
#. We are now getting close to installing pysatCDF. Do the following in the
   shell environment that is already opened.::

		mkdir src
		cd src

> 		git clone https://github.com/rstoneback/pysatCDF.git

Reviewer: ```suggestion
		git clone https://github.com/pysat/pysatCDF.git
```

		cd pysatCDF

#. Using a text editor of your choice, create a file called setup.cfg in::

		C:\msys64\home\gayui\src\pysatCDF

   Put the following in the file before saving and closing it.::

		[build]
		compiler=mingw32

   .. note::

>        gayui will be replaced with your username

Reviewer: ```suggestion
       gayui will need to be replaced with your username
```
